File: kotlin/math/main_9375.kt
fun main() {
    val count = readLine()!!.toInt()
    for (i in 1..count){
        val num = readLine()!!.toInt()
        val clothes = mutableListOf<String>()
        for (j in 1..num){
            clothes.add(readLine()!!.split(' ')[1])
        }

        var ans = 1
        clothes.groupingBy { it }.eachCount().values.forEach {
            ans *= (it + 1)
        }
        println(ans - 1)
    }
}

File: cpp/math/main_9375.cpp
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;

int main() {
	int count;
	cin >> count;
	for (int i = 0; i < count; ++i) {
		int num;
		cin >> num;
		map<string, int> clothes;
		for (int i = 0; i < num; ++i) {
			string name, type;
			cin >> name >> type;
			if (clothes.find(type) != clothes.end()) {
				clothes[type]++;
			} else {
				clothes[type] = 1;
			}
			
		}
		int ans = 1;
		for (auto && cloth : clothes) {
			ans *= (cloth.second + 1);
		}
		cout << ans - 1 << '\n';
	}
}